spharpy/samplings/coordinates.py

```diff
@@ -303,3 +354,36 @@ def n_max(self, value):
             self._n_max = None
         else:
             self._n_max = int(value)
+
+    @property
+    def radius_tolerance(self):
+        """Get or set the radius tolerance in meter."""
+        return self._radius_tolerance
+
+    @radius_tolerance.setter
+    def radius_tolerance(self, value):
+        """Get or set the radius tolerance in meter."""
+        if isinstance(value, (int, float)) and value > 0:
+            self._radius_tolerance = float(value)
+        else:
+            raise ValueError(
+                'The radius tolerance must be a number greater than zero')
+
+    def _check_points(self, x, y, z):
+        """Check input data before setting coordinates"""
+
+        # convert to numpy arrays of the same shape
+        x, y, z = super()._check_points(x, y, z)
+
+        # check for equal radius
+        radius = np.sqrt(x.flatten()**2 + y.flatten()**2 + z.flatten()**2)
+        radius_delta = np.max(np.abs(np.mean(radius) - radius))
```

Comment:
should we check here, between the largest and the smallest radius?

I found your example in the test with 0,0,0 and 1,0,0 and the error is returning 0.5m which does not really mean anything.
also using mean or median, could lead to different behaviour here. The distance between the lowest and largest radius is more meaningful in my point of view.

I would suggest the largest distance beween the radius, so max-min

Reply:
I did this as suggested